src/tests/unit/services/YotiCallbackProcessor.test.ts

```diff
@@ -507,46 +623,1016 @@ describe("YotiCallbackProcessor", () => {
+					 "credentialSubject":{
+						"name":[
+								 {
+								"nameParts":[
+											 {
+										"value":"TECH",
+										"type":"GivenName",
+											 },
+											 {
+										"value":"REFRESH",
+										"type":"GivenName",
+											 },
+											 {
+										"value":"ICTHREEMALE",
+										"type":"FamilyName",
+											 },
+								],
+								 },
+						],
+						"birthDate":[
+								 {
+								"value":"1988-12-04",
+								 },
+						],
+						"residencePermit":[
+								 {
+									"documentNumber": "RF9082242",
+									"expiryDate": "2024-11-11",
+									"issueDate": "2015-05-19",
+									"icaoIssuerCode": "GBR"
+								 },
+						],
+					 },
+					 "evidence":[
+						{
+								 "type":"IdentityCheck",
+								 "strengthScore":3,
+								 "validityScore":2,
+								 "verificationScore":3,
+								 "checkDetails":[
+								{
+											 "checkMethod":"vri",
+											 "txn":"b988e9c8-47c6-430c-9ca3-8cdacd85ee91",
+											 "identityCheckPolicy":"published",
+								},
+								{
+											 "checkMethod":"pvr",
+											 "txn":"b988e9c8-47c6-430c-9ca3-8cdacd85ee91",
+											 "photoVerificationProcessLevel":3,
+								},
+								 ],
+						},
+					 ],
+				},
+		 })],
+		});
+		expect(out.statusCode).toBe(HttpCodesEnum.OK);
+		expect(out.body).toBe("OK");
+		jest.useRealTimers();
+		jest.clearAllMocks();
+	});
+
+	it("Return successful response with 200 OK when YOTI session created with EU driving permit", async () => {
+		documentFields = getEuDrivingPermitFields();
+		const euDLYotiSession = getCompletedYotiSession();
+		euDLYotiSession.resources.id_documents[0].document_type = "DRIVING_LICENCE"
+		euDLYotiSession.resources.id_documents[0].issuing_country = "DEU"
+		jest.useFakeTimers();
+		jest.setSystemTime(absoluteTimeNow());
+		mockYotiService.getCompletedSessionInfo.mockResolvedValueOnce(euDLYotiSession);
+		mockYotiService.getMediaContent.mockResolvedValueOnce(documentFields);
+		mockF2fService.getSessionByYotiId.mockResolvedValueOnce(f2fSessionItem);
+
+		// @ts-ignore
+		mockYotiCallbackProcessor.verifiableCredentialService.kmsJwtAdapter = passingKmsJwtAdapterFactory();
+
+		const out: Response = await mockYotiCallbackProcessor.processRequest(VALID_REQUEST);
+
+		// eslint-disable-next-line @typescript-eslint/unbound-method
+		expect(mockF2fService.sendToTXMA).toHaveBeenCalledTimes(2);
+		// eslint-disable-next-line @typescript-eslint/unbound-method
+		expect(mockF2fService.sendToTXMA).toHaveBeenNthCalledWith(1, {"client_id": "ipv-core-stub", "component_id": "https://XXX-c.env.account.gov.uk", "event_name": "F2F_YOTI_RESPONSE_RECEIVED", "timestamp": absoluteTimeNow(), "user": {"govuk_signin_journey_id": "sdfssg", "ip_address": "127.0.0.1", "persistent_session_id": "sdgsdg", "session_id": "RandomF2FSessionID", "transaction_id": undefined, "user_id": "testsub", "email": undefined}});
```

Comment:
These data structures are massive and there is a fair amount of duplication. Do you think you could extract them into data files and reuse parts that can be reused? Happy to pair on it if that's helpful!

Reply:
Some refactoring has been done, a ticket has been raised to address as soon as we get spare time https://govukverify.atlassian.net/browse/F2F-919